### PR TITLE
[ci] chore: npu ci use cann9.0.0

### PR DESCRIPTION
**Pull request by daikang6**
### What does this PR do?

> update npu's cann to 9.0.0 version

### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: ...
- [ ] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `veomni`, `sglang`, `vllm`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`, `cfg`, `reward`, `fully_async`, `one_step_off`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/verl-project/verl/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/verl-project/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/verl-project/verl/tree/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/verl-project/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [ ] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ). (If not accessible, please try [the Feishu group (飞书群)](https://applink.larkoffice.com/client/chat/chatter/add_by_link?link_token=772jd4f1-cd91-441e-a820-498c6614126a).)
- [ ] If your PR is related to the `recipe` submodule, please also update the reference to the submodule commit via `git submodule update --remote` or `cd recipe && git pull origin main`.

### CODE REVIEW

File: .github/workflows/e2e_ascend.yml [hunk 1]
```diff
@@ -62,15 +62,15 @@ concurrency:
 
 permissions:
   contents: read
-
+
```
Comment (daikang6) [inline .github/workflows/e2e_ascend.yml:65]: npu unit test加上
import verl.utils.device as device_module
    monkeypatch.setattr(device_module, "is_npu_available", False)

File: .github/workflows/e2e_ascend.yml [hunk 2]
```diff
@@ -62,15 +62,15 @@ concurrency:
...
+ 
 jobs:
   llm_rl_job:
     if: github.repository_owner == 'verl-project'
     name: E2E Ascend testing for RL training scenarios of LLM models
     runs-on: linux-aarch64-a3-8
     timeout-minutes: 120
     container:
-      image: swr.cn-southwest-2.myhuaweicloud.com/modelfoundry/ascend-ci/verl/verl:verl-8.5.0-a3-ubuntu22.04-py3.11-latest
+      image: swr.cn-southwest-2.myhuaweicloud.com/modelfoundry/ascend-ci/verl/verl:verl-9.0.0-a3-ubuntu22.04-py3.11-latest
```
Comment (daikang6) [inline .github/workflows/e2e_ascend.yml:73]: # 获取专家总数（从 fused 参数第 0 维读取，兼容 Qwen3MoE）
        hf_experts = hf_layer.mlp.experts
        num_experts = getattr(hf_experts, 'num_experts', None) or hf_experts.gate_up_proj.shape[0]

        num_local_experts = num_experts // ep_size
        expert_idx_start = ep_rank * num_local_experts
        expert_idx_end = (ep_rank + 1) * num_local_experts

        # 适配 transformers 5.x Qwen3MoE：gate_up_proj + down_proj 为三维张量
        if hasattr(hf_experts, 'gate_up_proj'):
            for idx in range(num_experts):
                if idx < expert_idx_start or idx >= expert_idx_end:
                    continue
                local_expert_idx = idx - expert_idx_start

                # gate_up_proj: [num_experts, 2 * intermediate_size, hidden_size]
                gate_up = hf_experts.gate_up_proj[idx]              # [2*I, H]
                intermediate_size = gate_up.shape[0] // 2
                gate_w = gate_up[:intermediate_size]                # [I, H]
                up_w   = gate_up[intermediate_size:]                # [I, H]

                fc1_weight = torch.cat([gate_w, up_w], dim=0)       # [2*I, H]
                # down_proj: [num_experts, hidden_size, intermediate_size]
                down_w = hf_experts.down_proj[idx]                  # [H, I]

                numel += safe_copy(fc1_weight, layer.mlp.experts.linear_fc1._parameters[f"weight{local_expert_idx}"])
                numel += safe_copy(down_w, layer.mlp.experts.linear_fc2._parameters[f"weight{local_expert_idx}"])

        # 兼容旧版 transformers / 其他 MoE（ModuleList 形式）
        elif hasattr(hf_experts, '__iter__'):
            for idx, hf_expert in enumerate(hf_experts):
                if idx < expert_idx_start or idx >= expert_idx_end:
                    continue
                local_expert_idx = idx - expert_idx_start

                fc1_weight = torch.cat([hf_expert.gate_proj.weight, hf_expert.up_proj.weight])
                numel += safe_copy(fc1_weight, layer.mlp.experts.linear_fc1._parameters[f"weight{local_expert_idx}"])
                numel += safe_copy(
                    hf_expert.down_proj.weight, layer.mlp.experts.linear_fc2._parameters[f"weight{local_expert_idx}"]
                )
        else:
            raise TypeError(f"Unsupported experts type: {type(hf_experts)}")
升级后，权重转换部分要这样改

File: tests/utils/test_megatron_bshd_preprocess.py [hunk 3]
```diff
@@ -39,6 +41,7 @@ def _load_mcore_util_with_stubbed_megatron(monkeypatch, tp_size: int = 4):
     monkeypatch.setitem(sys.modules, "megatron.core", core)
     monkeypatch.setitem(sys.modules, "megatron.core.parallel_state", parallel_state)
     monkeypatch.setitem(sys.modules, "megatron.core.packed_seq_params", packed_seq_params)
+    monkeypatch.setattr(device_module, "is_npu_available", False)
```
Comment (daikang6) [inline tests/utils/test_megatron_bshd_preprocess.py:44]: pytest -s -x --ignore-glob="*test_special_*.py" --ignore-glob="*on_cpu.py" --ignore-glob="*test_vllm*" --ignore-glob="*_sglang*" --ignore-glob="*_hf_rollout*" --ignore-glob="tests/models/" --ignore-glob="tests/special*" --ignore-glob="tests/experimental" --ignore-glob="tests/workers/reward_model" --ignore-glob="*test_rvdz*" --ignore-glob="*test_ray_collectives*" --ignore-glob="*test_nvtx_profile*" --ignore-glob="tests/checkpoint_engine" --ignore-glob="*test_shared_memory*" --ignore-glob="tests/workers/rollout/rollout_trtllm" --ignore-glob="*test_fsdp_lora_merge*" --ignore-glob="*test_activation_offload*" --ignore-glob="*test_normalize_peft_param_name.py*" tests/ -k "not test_preprocess_bshd_engine_preserves_topk_dense_dim_on_gpu"    skip这个用例

**Review by wucong25 — APPROVED** [review]
